File: src/main/java/io/github/takenoha/towerdefense/persistence/RaidSealRefundResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Result of the one-time technical refund of a consumed raid seal. */
public record RaidSealRefundResult(OperationOutcome outcome, RaidSeal returnedSeal) {
    public RaidSealRefundResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(returnedSeal, "returnedSeal");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/RaidSealRefundResult.kt
package io.github.takenoha.towerdefense.persistence

import kotlin.jvm.JvmRecord

/** Result of the one-time technical refund of a consumed raid seal. */
@JvmRecord
data class RaidSealRefundResult(
    val outcome: OperationOutcome,
    val returnedSeal: RaidSeal,
)
